### PR TITLE
Add append tests for wait-on-conflict concurrency control

### PR DESCRIPTION
**Pull request by qvad**

### CODE REVIEW

File: yugabyte/src/yugabyte/auto.clj [hunk 1]
```diff
@@ -322,13 +321,23 @@
...
   [test]
-  (if (and (= (:workload test) :ysql/append-rc))
+  (if (contains? [:ysql/rc.ol.append :ysql/rc.pl.append] (:workload test))
     [:--yb_enable_read_committed_isolation]
     []))
 
+(defn master-tserver-pessimistic-locking-flags
+  "Version-specific flags"
+  [test]
+  (if (contains? [:ysql/rc.pl.append
```
Comment (frozenspider) [inline yugabyte/src/yugabyte/auto.clj]: Maybe just ensure workload name contains `.pl.` ?

File: yugabyte/src/yugabyte/ysql/append.clj [hunk 2]
```diff
@@ -36,30 +36,45 @@
...
-  (some-> conn
-          (c/query [(str "select (" col ") from " table " where k = ?") row])
-          first
-          (get (keyword col))
-          (str/split #",")
-          (->>                                              ; Append might generate a leading , if the row already exists
-            (remove str/blank?)
-            (mapv #(Long/parseLong %)))))
+  [locking conn table row col]
+    (some-> conn
```
Comment (frozenspider) [inline yugabyte/src/yugabyte/ysql/append.clj]: Not sure if this indentation change is in line with the style, probably not.

**Review by frozenspider — APPROVED** [review]
Accepted with comments

File: yugabyte/src/yugabyte/ysql/append.clj [hunk 3]
```diff
@@ -36,11 +36,24 @@
   [test k]
   (str "v" (mod k keys-per-row)))
 
+(defn select-with-pessimistic
```
Comment (robertsami) [inline yugabyte/src/yugabyte/ysql/append.clj]: maybe more appropriate to call this `select-with-lock` as the clauses used here are available/valid even when wait-on-conflict behavior is disabled

File: yugabyte/src/yugabyte/ysql/append.clj [hunk 4]
```diff
@@ -36,11 +36,24 @@
...
+  [locking col table]
+  (let [clause (if (= :pessimistic locking)
+                 (rand-nth ["" " for update" " for no key update" " for share" " for key share"])
+                 "")]
+    (str "select (" col ") from " table " where k = ?" clause)))
+
+(defn lock-row-if-needed
+  [locking conn col table v row]
+  (if (= :pessimistic locking)
+    (c/execute! conn [(select-with-pessimistic locking col table) v row]))
```
Comment (robertsami) [inline yugabyte/src/yugabyte/ysql/append.clj]: why do we need to do a select with lock before each update if we're testing wait-on-conflict concurrency control?

File: yugabyte/src/yugabyte/auto.clj [hunk 5]
```diff
@@ -413,7 +421,8 @@
             :--enable_tracing
             :--rpc_slow_query_threshold_ms 1000
             (tserver-api-opts (:api test) node)
-            (tserver-workload-specific-opts test)
+            (tserver-read-committed-flags test)
+            (master-tserver-pessimistic-locking-flags test)
```
Comment (robertsami) [inline yugabyte/src/yugabyte/auto.clj]: are we still running it /without/ wait on conflict?

also can we rename this `master-tserver-wait-on-conflict-flags`?

**Review by robertsami — CHANGES_REQUESTED** [review]

**Review by robertsami — APPROVED** [review]
LGTM once comments are addressed

**Pull request merged**